tests/dag_test.cpp

```diff
@@ -227,6 +227,84 @@ TEST_F(DagTest, compute_epoch) {
+  DagBlock blkF(blk_hash_t(3), 1, {}, {}, sig_t(1), blk_hash_t(7), addr_t(1));
+  DagBlock blkG(blk_hash_t(2), 1, {}, {trx_hash_t(4)}, sig_t(1), blk_hash_t(8), addr_t(1));
+  DagBlock blkH(blk_hash_t(6), 4, {blk_hash_t(8), blk_hash_t(10)}, {}, sig_t(1), blk_hash_t(9), addr_t(1));
+  DagBlock blkI(blk_hash_t(11), 3, {blk_hash_t(4)}, {}, sig_t(1), blk_hash_t(10), addr_t(1));
+  DagBlock blkJ(blk_hash_t(7), 2, {}, {}, sig_t(1), blk_hash_t(11), addr_t(1));
+  DagBlock blkK(blk_hash_t(9), 5, {}, {}, sig_t(1), blk_hash_t(12), addr_t(1));
+
+  const auto blkK_hash = blkK.getHash();
+
+  mgr->addDagBlock(std::move(blkA));
+  mgr->addDagBlock(std::move(blkB));
+  mgr->addDagBlock(std::move(blkC));
+  mgr->addDagBlock(std::move(blkD));
+  mgr->addDagBlock(std::move(blkF));
+  mgr->addDagBlock(std::move(blkE));
+  mgr->addDagBlock(std::move(blkG));
+  mgr->addDagBlock(std::move(blkJ));
+  mgr->addDagBlock(std::move(blkI));
+  mgr->addDagBlock(std::move(blkH));
+  mgr->addDagBlock(std::move(blkK));
+  taraxa::thisThreadSleepForMilliSeconds(100);
+
+  vec_blk_t orders;
+  orders = mgr->getDagBlockOrder(blkK_hash, 1);
+
+  // Verify all blocks made it to DAG
+  EXPECT_EQ(orders.size(), 11);
+
+  // Verify initial expiry level
+  EXPECT_EQ(mgr->getDagExpiryLevel(), 0);
+  mgr->setDagBlockOrder(blkK_hash, 1, orders);
+
+  // Verify expiry level
+  EXPECT_EQ(mgr->getDagExpiryLevel(), blkK.getLevel() - EXPIRY_LIMIT);
+
+  DagBlock blk_under_limit(blk_hash_t(2), blkK.getLevel() - EXPIRY_LIMIT - 1, {}, {}, sig_t(1), blk_hash_t(13),
+                           addr_t(1));
+  DagBlock blk_at_limit(blk_hash_t(4), blkK.getLevel() - EXPIRY_LIMIT, {}, {}, sig_t(1), blk_hash_t(14), addr_t(1));
+  DagBlock blk_over_limit(blk_hash_t(11), blkK.getLevel() - EXPIRY_LIMIT + 1, {}, {}, sig_t(1), blk_hash_t(15),
+                          addr_t(1));
+
+  // Block under limit is not accepted to DAG since it is expired
+  EXPECT_FALSE(mgr->addDagBlock(std::move(blk_under_limit)));
+  EXPECT_TRUE(mgr->addDagBlock(std::move(blk_at_limit)));
+  EXPECT_TRUE(mgr->addDagBlock(std::move(blk_over_limit)));
+  EXPECT_FALSE(db_ptr->dagBlockInDb(blk_under_limit.getHash()));
+  EXPECT_TRUE(db_ptr->dagBlockInDb(blk_at_limit.getHash()));
+  EXPECT_TRUE(db_ptr->dagBlockInDb(blk_over_limit.getHash()));
+
+  DagBlock blk_new_anchor(blk_hash_t(12), 6, {}, {}, sig_t(1), blk_hash_t(16), addr_t(1));
+  EXPECT_TRUE(mgr->addDagBlock(std::move(blk_new_anchor)));
+
+  orders = mgr->getDagBlockOrder(blk_new_anchor.getHash(), 2);
+  mgr->setDagBlockOrder(blk_new_anchor.getHash(), 2, orders);
+
+  // Verify that the block blk_at_limit which was initially part of the DAG became expired once new anchor moved the
+  // limit
+  EXPECT_FALSE(db_ptr->dagBlockInDb(blk_under_limit.getHash()));
+  EXPECT_FALSE(db_ptr->dagBlockInDb(blk_at_limit.getHash()));
+  EXPECT_TRUE(db_ptr->dagBlockInDb(blk_over_limit.getHash()));
+}
```

Comment:
Could you please add tests for clearPeriodDataHistory? Not remove non-expired DAG blocks

Reply:
I have added test for this FullNodeTest clear_period_data